apps/ui/src/components/textfield/__stories__/helpers.tsx

```diff
@@ -0,0 +1,80 @@
+import { useState, useEffect, useCallback } from 'react';
+import type { ComponentProps, ReactNode } from 'react';
+import { TextField as TextFieldComponent } from '../text-field';
+
+export function useControlledString(initial = '') {
+  const [value, setValue] = useState(initial);
+  return {
+    value,
+    onChange: setValue,
+    setValue,
+  };
+}
+
+export function TextFieldWithState(props: ComponentProps<typeof TextFieldComponent>) {
```

Comment:
I recommend removing this `TextFieldWithState` helper; it should be implemented in the component itself.
The component will be a either **controlled** & **uncontrolled** component